### ISSUE TITLE
Is there a way to persist data across updates? 

### ISSUE THREAD

**Issue by dlcole** [EddyVerbruggen/nativescript-secure-storage#38]
It looks like there used to be, at least on iOS.  Is it possible on either platform?

**Comment by hgc2002** [comment]
Because NS clear all the app folder when a new version is installed, it looks impossible.

**Comment by hgc2002** [comment]
Workaround: use external storage. Apparently, it's the right way to do it (and the only one supported everywhere without crossing any safety line).
I'm using it right now and it goes well.

**Comment by spstratis** [comment]
Hi, when you say updates here, were you meaning application updates or Phone OS updates?

**Comment by dlcole** [comment]
I was referring to application updates.  

I don't know that you can depend on external storage being available, especially on iOS.

**Comment by spstratis** [comment]
I see, so it's confirmed that keychain data for a Nativescript app gets cleared out after application updates? Because I believe I'm seeing similar behavior in one of my applications.

**Comment by dlcole** [comment]
No, I'm not sure that's confirmed.  I had understood that keychain *is* a way you can preserve data across application updates, and the plugin includes a method to clear data on first run.  I never really pursued using this plugin but may investigate it further in the future.

**Comment by madmas** [comment]
@dlcole the data is persisted during updates. I've used this plugin and the data is not lost during updates. For future usage one should consider #40 and maybe contribute to update to a more future-proof implementation.

**Comment by UNIcodeX** [comment]
How would one set the save location for nativescript-secure-storage to the "external" area, be it emulated or real??

I'm currently using this library in a project and the data comes back across uninstalls and updates in iOS like I want it to, but is not persisting across Android updates.

Currently there is really no need to save this data across device changes and so persistence would be awesome.

**Comment by darko-bw** [comment]
Totally agree with @UNIcodeX
